core/core_integrations/slack.js

```diff
@@ -17,596 +15,553 @@ inTransaction = {},
+                break;
+            case 'pong':
+                socket.pong();
+                break;
+            default:
+                socket.send(body);
+        }
+    }
+    else {
+        console.debug('slack-> No socket available for given team id');
+    }
 },
 
-sendSocketMessage = (teamId, body, callback) => {
-	let socket = sockets[teamId];
-	if (socket !== null) {
-		socket.send(body, undefined, (response) => {
-			if(response) {
-				console.debug('slack-> Error on socket sending message, error: ' + response);
-			}
-			if (callback) {
-				callback(response);
-			}
-		});
-	}
-	else {
-		console.debug('slack-> No socket available for given team id');
-	}
+sendTyping = function(thread) {
+    var teamInfo = getChannelIdAndTeamId(thread),
+        teamId = teamInfo.team_id,
+        body = {
+            type:'typing',
+            channel: teamInfo.channel_id
+        };
+
+    body.id = getNextMessageId(teamId);
+    body = JSON.stringify(body);
+
+    console.debug('slack-> send typing indication');
+    sendSocketMessage('message', teamId, body);
 },
 
 replaceUserIdWithUserName = function(userName, message, match) {
-	if (userName) {
-		var index = message.indexOf(match);
-		return message.substr(0, index) + userName + message.substr(index + match.length);
-	}
+    if (userName) {
+        var index = message.indexOf(match);
+        return message.substr(0, index) + userName + message.substr(index + match.length);
+    }
 },
 
-init = function(token, callback) {
-	var body = {
-		"token": token,
-		"mpim_aware": true
-	};
-	request({
-		"uri": 'https://slack.com/api/rtm.start',
-		"method": 'GET',
-		"qs": body
-	},
-	function (error, response, body) {
-		if (error) {
-			console.debug("slack-> failed to Initialise connection: " + error);
-			callback(false);
-			return;
-		}
-		var teamId = null,
-		slackTeams = exports.config.slack_teams,
-		team;
-
-		if (!slackTeams) {
-			exports.config.slack_teams = {};
-			slackTeams = exports.config.slack_teams;
-		}
-
-		try {
-			body = JSON.parse(body);
-		}
-		catch(e) {
-			console.debug("slack-> failed to parse message: " + body);
-			console.critical(e);
-			callback(false);
-			return;
-		}
-
-		if (response.statusCode !== 200) {
-			console.debug('slack-> error: ' + response.statusCode);
-			callback(false);
-		}
-		else if (!body.ok) {
-			console.debug('slack-> Failed to connect to slack server with token: ' + token + ' error: ' + body.error);
-			callback(false);
-		}
-		else {
-			teamId = body.team.id;
-			slackTeams[teamId] = body;
-			team = slackTeams[teamId];
-			team.token = token;
-			team.event_id = 1;
-			team.lastMessageSinceConnection = false;
-
-			//Generate user map
-			team.users = generateUserMap(body);
-			team.allChannels = generateChannelsMap(body);
-
-			tempTeamData[teamId] = {};
-
-			// find slackbot
-			findSlackBot(body, team);
-			callback({"team_id": teamId, "url": body.url});
-		}
-	});
-},
+generateChannelsMap = function(data) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l177)